src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -938,12 +938,13 @@ void MacroAssembler::long_move(VMRegPair src, VMRegPair dst) {
         mov(dst.first()->as_Register(), src.first()->as_Register());
       }
     } else {
-      assert(dst.is_single_reg(), "not a stack pair");
+      assert(dst.is_single_reg(), "not a stack pair: (%s, %s), (%s, %s)",
+       src.first()->name(), src.second()->name(), dst.first()->name(), dst.second()->name());
       movq(Address(rsp, reg2offset_out(dst.first())), src.first()->as_Register());
     }
   } else if (dst.is_single_phys_reg()) {
     assert(src.is_single_reg(),  "not a stack pair");
-    movq(dst.first()->as_Register(), Address(rbp, reg2offset_out(src.first())));
+    movq(dst.first()->as_Register(), Address(rbp, reg2offset_in(src.first())));
```

Comment:
A pre-existing bug found when testing (same below). Loads from `rbp` should always be paired with `reg2offset_in`